src/LogikUI/File/FileManager.cs

```diff
@@ -0,0 +1,337 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml;
+using System.Xml.Schema;
+using System.Xml.XPath;
+using System.IO;
+using Gtk;
+using LogikUI.Util;
+using LogikUI.Circuit;
+using LogikUI.Simulation.Gates;
+
+namespace LogikUI.File
+{
+    /* FILE STRUCTURE:
+     *  circuit
+     *    wires
+     *      wire
+     *        from < -x: int, y: int
+     *        to < -x: int, y: int
+     *      ...
+     *    components
+     *      component
+     *        type
+     *          "buffer" | "and" | "or" | "xor"
+     *        location < -x: int, y: int
+     *        orientation
+     *          "north" | "east" | "south" | "west"
+     *      ...
+     *    labels
+     *      label < -size: int(10, 100)
+     *        location < -x: int, y: int
+     *        text
+     *          <string>
+     *      ...
+     */
+
+    /// <summary>
+    /// Handles writing and parsing of project files.
+    /// </summary>
+    static class FileManager
+    {
+        private static Dictionary<string, ComponentType> types = new Dictionary<string, ComponentType>()
+        {
+            { "buffer", ComponentType.Buffer },
+            { "and", ComponentType.And },
+            { "or", ComponentType.Or },
+            { "xor", ComponentType.Xor },
+        };
```

Comment:
This dictionary feels like it's going to be a maintainability mess where every time a component is added it needs to be added to this list.
I should like to see this changed to using either just the integer value of the enum (these won't change), or use the `ToString` and `Enum.Parse` function (this is less desired as the enum names can't be changed then).